### PR TITLE
Emagged clown cars now come with siege-tank mode.

### PR DESCRIPTION
**Pull request by ghost**
:cl: Floyd / Qustinnus (thx to mrdoombringer for sprite + idea)
add: The clowncar now comes with an inbuilt cannon that can be activated by emagging its circuits
fix: cars can only kidnap if they have the CAN_KIDNAP trait
fix: removes a dot too much in a to_chat in clown car actions
/:cl:

lol i'll run out of screen space for all these buttons soon
https://streamable.com/yq17r

### CODE REVIEW

**Comment by zxaber** [comment]
Now we just need mime-themed zerglings

**Comment by Qustinnus** [comment]
btw before anyone asks again, im not nerfing tactical resting

**Comment by GunHog** [comment]
### ADVANCED CLOWN TACTICS

**Comment by Bawhoppen** [comment]
You've been doing really well with this clown car feature. Keep up the good work! :+1:

**Comment by ExcessiveUseOfCobblestone** [comment]
emag function...

**Comment by GuyonBroadway** [comment]
Now make it optionally shoot pies.

**Comment by MimicFaux** [comment]
Idea: If the car has no more captives, it shoots the driver.
Now I know what you're thinking, did he fire six assistants, or only five?
Well to tell you the truth, in all this honking, I kinda lost track myself.
but being that this is a 44 crew cannon, the most powerful launcher on the station, you gotta ask yourself.
Do you feel lucky, cap?

**Comment by GunHog** [comment]
Launching yourself from a cannon is a popular daredevil act, after all...

**Comment by GuyonBroadway** [comment]
@mimicfaux 

Better idea, give the car a dedicated BUTTON to fire yourself. Either via cannon where you aim or directly forwards when in car mode. 

Useful? Hardly.
Fun? Absolutley!

Also waiting for the day the clown hits the ejector and quickly uses a chameleon projector to evade the law.

**Comment by MrDoomBringer** [comment]
i just realized i never sent qustinnus the sprites for this oops
i had em sitting on me compuesto for a worryingly long time
he said he'll PR em tomorrorw

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -877,6 +877,12 @@
 		var/obj/mecha/M = loc
 		if(M.mouse_pointer)
 			client.mouse_pointer_icon = M.mouse_pointer
+	if (istype(loc, /obj/vehicle/sealed))
```
Comment (Cyberboss) [inline code/modules/mob/mob.dm]: else if

File: code/modules/vehicles/cars/clowncar.dm [hunk 2]
```diff
@@ -128,3 +136,44 @@
...
+	if(cannonmode)
+		cannonmode = FALSE
+		flick("clowncar_fromfire", src)
+		cannonbusy = TRUE
+		addtimer(CALLBACK(src, .proc/LeaveCannonMode), 20)
+		playsound(src, 'sound/vehicles/clowncar_cannonmode2.ogg', 75)
+		visible_message("<span class='danger'>The [src] starts going back into mobile mode.</span>")
+	else
+		canmove = FALSE
+		cannonbusy = TRUE
```
Comment (Cyberboss) [inline code/modules/vehicles/cars/clowncar.dm]: duplicated line

File: code/modules/vehicles/sealed.dm [hunk 3]
```diff
@@ -1,5 +1,6 @@
 /obj/vehicle/sealed
 	var/enter_delay = 20
+	var/mouse_pointer = null
```
Comment (Cyberboss) [inline code/modules/vehicles/sealed.dm]: remoof null

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/vehicles/sealed.dm (hunk 3 above)
Comment (MrDoomBringer) [reply code/modules/vehicles/sealed.dm]: >living in 2018 and still doing this

qusty wyd

Comment (Qustinnus) [reply code/modules/vehicles/sealed.dm]: i think i copy pasted the line from mecha code.

tbf i shouldve known going into mecha code

File: code/modules/vehicles/cars/clowncar.dm (hunk 2 above)
Comment (Qustinnus) [reply code/modules/vehicles/cars/clowncar.dm]: wdym

File: code/modules/vehicles/sealed.dm (hunk 3 above)
Comment (Cyberboss) [reply code/modules/vehicles/sealed.dm]: >living in 2018 and still not following code standards

Comment (Qustinnus) [reply code/modules/vehicles/sealed.dm]: reeeeee

Comment (Qustinnus) [reply code/modules/vehicles/sealed.dm]: atleast you cant get into the clowncar without having the key >:(

**Comment by Qustinnus** [comment]
image

**Pull request merged**